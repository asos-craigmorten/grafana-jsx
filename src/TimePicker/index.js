/**
 * REF: https://grafana.com/docs/grafana/latest/reference/dashboard/#timepicker
 */

import PropTypes from "prop-types";

const TimePicker = ({
  collapse = false,
  enable = true,
  notice = false,
  now = true,
  refreshIntervals = [
    "5s",
    "10s",
    "30s",
    "1m",
    "5m",
    "15m",
    "30m",
    "1h",
    "2h",
    "1d",
  ],
  timeOptions = ["5m", "15m", "1h", "6h", "12h", "24h", "2d", "7d", "30d"],
  status = "Stable",
  type = "timepicker",
  ...props
}) => (
  <timepicker
    collapse={collapse}
    enable={enable}
    notice={notice}
    now={now}
    refresh_intervals={refreshIntervals}
    status={status}
    time_options={timeOptions}
    type={type}
    {...props}
  />
);

TimePicker.PropTypes = {
  collapse: PropTypes.bool,
  enable: PropTypes.bool,
  notice: PropTypes.bool,
  now: PropTypes.bool,
  refreshIntervals: PropTypes.arrayOf(PropTypes.string),
  status: PropTypes.string,
  type: PropTypes.string,
};

export default TimePicker;
