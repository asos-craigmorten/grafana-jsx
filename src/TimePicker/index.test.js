import TimePicker from "./";

const withAllProps = {
  collapse: true,
  enable: false,
  notice: true,
  now: false,
  refreshIntervals: ["test-refresh-interval-1", "test-refresh-interval-2"],
  timeOptions: ["test-time-option-1", "test-time-option-2"],
  status: "test-status",
  type: "test-type",
};

describe("TimePicker", () => {
  describe("when used in isolation", () => {
    describe("when passed no props", () => {
      it("should render the default timepicker object", () => {
        expect(<TimePicker />).toMatchSnapshot();
      });
    });

    describe("when passed all props", () => {
      it("should render a custom timepicker object", () => {
        expect(<TimePicker {...withAllProps} />).toMatchSnapshot();
      });
    });
  });

  describe("when used within another object", () => {
    describe("when passed no props", () => {
      it("should add a default timepicker object to the outer object", () => {
        expect(
          <test>
            <TimePicker />
          </test>
        ).toMatchSnapshot();
      });
    });

    describe("when passed all props", () => {
      it("should add a custom timepicker object to the outer object", () => {
        expect(
          <test>
            <TimePicker {...withAllProps} />
          </test>
        ).toMatchSnapshot();
      });
    });
  });
});
