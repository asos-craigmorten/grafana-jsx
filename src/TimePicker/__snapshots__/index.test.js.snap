// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker when used in isolation when passed all props should render a custom timepicker object 1`] = `
Object {
  "collapse": true,
  "enable": false,
  "notice": true,
  "now": false,
  "refresh_intervals": Array [
    "test-refresh-interval-1",
    "test-refresh-interval-2",
  ],
  "status": "test-status",
  "time_options": Array [
    "test-time-option-1",
    "test-time-option-2",
  ],
  "type": "test-type",
}
`;

exports[`TimePicker when used in isolation when passed no props should render the default timepicker object 1`] = `
Object {
  "collapse": false,
  "enable": true,
  "notice": false,
  "now": true,
  "refresh_intervals": Array [
    "5s",
    "10s",
    "30s",
    "1m",
    "5m",
    "15m",
    "30m",
    "1h",
    "2h",
    "1d",
  ],
  "status": "Stable",
  "time_options": Array [
    "5m",
    "15m",
    "1h",
    "6h",
    "12h",
    "24h",
    "2d",
    "7d",
    "30d",
  ],
  "type": "timepicker",
}
`;

exports[`TimePicker when used within another object when passed all props should add a custom timepicker object to the outer object 1`] = `
Object {
  "timepicker": Object {
    "collapse": true,
    "enable": false,
    "notice": true,
    "now": false,
    "refresh_intervals": Array [
      "test-refresh-interval-1",
      "test-refresh-interval-2",
    ],
    "status": "test-status",
    "time_options": Array [
      "test-time-option-1",
      "test-time-option-2",
    ],
    "type": "test-type",
  },
}
`;

exports[`TimePicker when used within another object when passed no props should add a default timepicker object to the outer object 1`] = `
Object {
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
}
`;
